### DISCUSSION TITLE
Are we going to release deprecated code in `1.0` releases

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3622] [General]
I believe we now have a soft policy in `rust-bitcoin` to keep deprecated code around indefinitely. What about for the upcoming `1.0` releases (`hashes`, `units`, `primitives` etc). Are we keeping deprecated in those?

FTR I rekon that we should not.

**Comment by shinghim** [comment]
I'm with you and think deprecated code shouldn't be kept indefinitely. It would cause the codebase to eventually become too bloated and hard to maintain. Major releases like `1.0` should be the ideal time to stop supporting deprecated code, though I'm not familiar/experienced enough with this codebase yet to say if that should go for everything. Maybe there are some exceptions of deprecated code that should be supported for a bit longer?

**Comment by apoelstra** [comment]
I vote yes, let's keep deprecated code. In almost all cases these deprecated functions are 1-3 lines of code and cost nearly nothing to maintain.

But it's not a strong feeling. If people want a clean API I can understand wanting to clear them out. In that case I'd like the offending functions to be in a 0.32.x release for at least 3 months to give people time to migrate.

**Reply by tcharding** [reply]
Oh I figured they stay in 0.32.x releases for ever

**Reply by tcharding** [reply]
(`bitcoin 0.32` will never depend on `units 1.0`, right?)

**Reply by apoelstra** [reply]
Yes -- they will be in 0.32.x forever. I just meant that when we release 1.0, we should keep anything that we only deprecated in 0.32.x very recently.

Though now that I say that, it seems a bit silly...there's no reason that users can't just delay their own upgrade. We will say in the 1.0 release notes that people should first upgrade to the 0.32.x and fix all the deprecation warnings.

**Comment by yancyribbens** [comment]
I don't know about the soft policy agreement, although I don't get the argument for keeping it around forever.  The deprecated code might all be small easy to maintain 3 liners or so, although that could change.
